### PR TITLE
refactor(hx-search-assistance): surf-1498 use positionable mixin

### PR DESCRIPTION
**Pull request by 100stacks**
* Refactored `HXSearchAssistance` to use the `Positionable` mixin.

JIRA: SURF-1498

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 9d85848d9beaf715acde065e3a2897d5cf68b739

https://app.netlify.com/sites/helix-ui/deploys/5c0edfbf05c4173093a1c861

File: docs/elements/hx-search-assistance/index.html [hunk 1]
```diff
@@ -42,10 +43,10 @@
 
 {% block attributes %}
   <dl>
-    <dt>open</dt>
+    <dt>open <i>(optional)</i></dt>
     <dd>Opens the search assistance</dd>
 
-    <dt>position {String [bottom-start]}</dt>
+    <dt>position {String [bottom-start]} <i>(optional)</i></dt>
```
Comment (CITguy) [inline docs/elements/hx-search-assistance/index.html]: Please remove `{String [bottom-start]}`

File: src/helix-ui/elements/HXSearchAssistanceElement.js [hunk 2]
```diff
@@ -1,181 +1,33 @@
...
-                element: this,
-                reference: this.relativeElement,
-                position: this.position,
-            });
-
-            this.style.top = `${y}px`;
-            this.style.left = `${x}px`;
-        }
+        super.$onConnect();
+        this.$defaultAttribute('role', 'search');
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchAssistanceElement.js]: Let's hold off on including the `role` attr. This means we'll have to remove the whole `$onConnect` method.

**Comment by 100stacks** [comment]
We'll need to address accessibility later during our Accessibility Audit.

Also, need to address this: https://github.com/rackerlabs/helix-ui/blob/master/src/helix-ui/mixins/Positionable.js#L150

**Pull request merged**